src/components/Dialogs/BuildDialog.tsx

```diff
@@ -0,0 +1,183 @@
+/**
+ * AlgoWorld Explorer
+ * Copyright (C) 2022 AlgoWorld
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import * as React from 'react';
+import Button from '@mui/material/Button';
+import Dialog from '@mui/material/Dialog';
+import DialogActions from '@mui/material/DialogActions';
+import DialogContent from '@mui/material/DialogContent';
+import DialogTitle from '@mui/material/DialogTitle';
+import { useMemo, useState } from 'react';
+import CryptoTextField from '../TextFields/CryptoTextField';
+import formatAmount from '@/utils/formatAmount';
+import { CoinType } from '@/models/CoinType';
+import { FROM_ASSET_PICKER_DIALOG_ID } from './constants';
+import { useAppSelector } from '@/redux/store/hooks';
+import { DialogContentText, Typography, Select, MenuItem, SelectChangeEvent } from '@mui/material';
+import { AWT_ASSET_ID } from '@/common/constants';
+import { MapAsset } from '@/models/MapAsset';
+
+type Props = {
+  onDepositConfirmed: (object: number) => void;
+  onDepositCancelled: () => void;
+  depositAsset: MapAsset;
+  open: boolean;
+};
+
+export const BuildDialog = ({
+  onDepositConfirmed,
+  onDepositCancelled,
+  depositAsset,
+  open,
+}: Props) => {
+  const assets = useAppSelector((state) => state.application.assets);
+  const chain = useAppSelector((state) => state.application.chain);
+
+  const [selectedobject, setSelectedobject] = useState('1');
+
+
+  const awtAsset = useMemo(() => {
+    const filteredAssets = assets.filter(
+      (asset) => asset.index === AWT_ASSET_ID(chain),
+    );
+    return filteredAssets.length === 1 ? filteredAssets[0] : undefined;
+  }, [assets, chain]);
+
+  const enough_awt = (formatAmount(awtAsset?.amount, awtAsset?.decimals) ?? 0) > depositAsset.cost
+  
+
+  const handleChange = (event:SelectChangeEvent) => {
+    setSelectedobject(event.target.value);
+  };
+
+  const renderSwitch = (param: number) => {
+    switch(param) {
+      case 1:
+        return 'Meadow';
```

Comment:
Please create an enum with value mappings to these types so you can instead pass param as enum and have comparisons like param === TextureType.WATER and etc. Lookups where enums are stored in the codebase and just define an extra enum there.

Reply:
Done :)